### DISCUSSION TITLE
Spend / Budget Cap for Nhost Platform (cloud)

### DISCUSSION THREAD

**Discussion by ammarfaris** [nhost/nhost#2846] [Ideas]
First and for most, we would like to say thanks and kudos to the Nhost's team for continuing to deliver - especially the two most awaited features which have been released => Rate Limiting - recently released in August 2024 & Official Support for React Native in July 2024 👏🏻

We would like to ask if another vital feature is in the roadmap and can be implemented please  => **Spend / Budget Cap for Nhost Platform (cloud)** ???

This feature is offered by some of Nhost's main competitors, and we hope that the feature can be implemented by Nhost too....

In the FAQ section of Nhost's pricing page, it says:

> What happens if I exceed the limits?
> We never stop your project and you will be charged for the excess usage. Pay as you grow.

On one side, it may be good to not disrupt a running application, but on the other side, developers are also worried if their spending / cost go over budget.

Hope Nhost can consider this feature and make it a priority.. thanks again 🙏🏻

**Comment by dbarrosop** [comment]
Hello,
thanks for your kind words. We are currently re-working billing to make it more convenient and flexible for users and one of the additions we are planning on adding is something along the lines you are requesting. We are not sure how it will look like but most likely we will add some form of email notification when certain user-defined thresholds are crossed. This way you can decide how to best move forward (ignore it/increase the cap, pause the project, remove some functions, lower resources, etc).

We haven't fully decided on how to do this yet so if you have any ideas don't hesitate to share so we can consider them when we start working on this.

If you have further questions/feedback don't hesitate to let us know.

Regards.

**Reply by ammarfaris** [reply]
Hey @dbarrosop, thanks again for your swift response - glad to know that your team is  re-working billing to improve it. 

I good start maybe is to look into something like [this ](https://supabase.com/docs/guides/platform/spend-cap#what-additional-usage-can-i-be-charged-for) ; where the spend / budget cap can be controlled item by item.

Thanks again, keep up the great work at Nhost!

**Reply by ammarfaris** [reply]
hi there, just want to check, will the implementation of Spend / Budget Cap for Nhost Platform in the near radar?

**Reply by dbarrosop** [reply]
https://docs.nhost.io/platform/cloud/billing#spending-notifications
